sdk/tables/azure-data-tables/inc/azure/data/tables/models.hpp

```diff
@@ -442,33 +442,121 @@ namespace Azure { namespace Data { namespace Tables {
     };
 
     /**
-     * @brief Table Entity
+     * @brief Table entity property.
      *
      */
-    struct TableEntity final
-    {
+    class TableEntityProperty final {
+    public:
+      /** Default constructor. */
+      TableEntityProperty() = default;
+
       /**
-       * Partition Key
+       * @brief Construct a new TableEntityProperty object.
+       *
+       * @param value Property value.
        */
-      std::string PartitionKey;
+      TableEntityProperty(std::string const& value) : Value(std::move(value)) {}
       /**
-       * Row Key
+       * @brief Construct a new TableEntityProperty object.
+       * @param value Property value.
+       * @param type Property type.
        */
-      std::string RowKey;
+      TableEntityProperty(std::string const& value, TableEntityDataType type)
+          : Value(std::move(value)), Type(type)
+      {
+      }
+      /**
+       * Property value.
+       */
+      std::string Value;
+      /**
+       * Property type.
+       */
+      Azure::Nullable<TableEntityDataType> Type;
+    };
+
+    /**
+     * @brief Table Entity
+     *
+     */
+    class TableEntity final {
+    public:
       /**
        * Properties
        */
-      std::map<std::string, std::string> Properties;
+      std::map<std::string, TableEntityProperty> Properties;
+
       /**
-       * ETag
+       * @brief Get partition key.
+       *
+       * @return Partition key
        */
-      Azure::Nullable<std::string> ETag;
+      TableEntityProperty GetPartitionKey() const { return GetProperty("PartitionKey"); }
 
       /**
-       * @brief Table Entity data type.
+       * @brief Set Partition key.
        *
+       * @param partitionKey Partition key.
        */
-      TableEntityDataType DataType;
+      void SetPartitionKey(const std::string& partitionKey)
+      {
+        Properties["PartitionKey"] = TableEntityProperty(partitionKey);
+      }
+
+      /**
+       * @brief Get row key.
+       *
+       * @return Row key
+       */
+      TableEntityProperty GetRowKey() const { return GetProperty("RowKey"); }
+      /**
+       * @brief Set Row key.
+       *
+       * @param rowKey Row key.
+       */
+      void SetRowKey(const std::string& rowKey)
+      {
+        Properties["RowKey"] = TableEntityProperty(rowKey);
```

Comment:
These manifest constants should probably be `constexpr const char *<name>=<value>;` to avoid possible errors with the name being used multiple times.

If they were only used in a single location it wouldn't be a big deal, but these are duplicated in the "Get" and "Set" methods.